internal/dns/records.go

```diff
@@ -2,93 +2,103 @@ package dns
 
 import (
 	"context"
+	"slices"
 
 	"github.com/miekg/dns"
 	"github.com/projectdiscovery/dnsx/libs/dnsx"
-)
-
-// Record represents a single DNS record.
-type Record struct {
-	Name  string `json:"name" yaml:"name"`
-	TTL   uint32 `json:"ttl" yaml:"ttl"`
-	Type  string `json:"type" yaml:"type"`
-	Value string `json:"value" yaml:"value"`
-}
 
-// Records represents all of the DNS records for a given domain.
-type Records struct {
-	A     []Record `json:"a" yaml:"a"`
-	AAAA  []Record `json:"aaaa" yaml:"aaaa"`
-	MX    []Record `json:"mx" yaml:"mx"`
-	TXT   []Record `json:"txt" yaml:"txt"`
-	NS    []Record `json:"ns" yaml:"ns"`
-	CNAME []Record `json:"cname" yaml:"cname"`
-}
-
-// RecordsReport represents the report of all DNS records for a given domain including all non-fatal errors that occurred.
-type RecordsReport struct {
-	Domain     string   `json:"domain" yaml:"domain"`
-	DNSRecords Records  `json:"dns_records" yaml:"dns_records"`
-	Errors     []string `json:"errors" yaml:"errors"`
-}
+	osintscan "github.com/Method-Security/osintscan/generated/go"
+)
 
-func getDNSRecords(domain string) (Records, error) {
+func getDNSRecords(domain string, questionTypes []uint16) (osintscan.DnsRecords, error) {
 	options := dnsx.DefaultOptions
-	var questionTypes []uint16 = []uint16{dns.TypeA, dns.TypeAAAA, dns.TypeMX, dns.TypeTXT, dns.TypeNS, dns.TypeCNAME}
 	options.QuestionTypes = questionTypes
-	options.MaxRetries = 5
```

Comment:
This is the default value, so we were specifying it redundantly. Could make it configurable for the user if we wanted to, but feels pretty in the weeds to be user facing unless we start running into concrete situations where downstream code would care about how many times we retry DNS queries